### PR TITLE
BaSP TG-74: Employees projects page

### PR DESCRIPTION
**Pull request by GuidoG95**
This page shows all the projects an employee is assinged to. A logged employee was mocked to show it's information because we don't have the login page yet. The employee mocked is the one with ID = 63756ccf76cb20ad5433fa22, but it can be changed modifying the mockedEmployeeLogged variable in line 24.

### CODE REVIEW

File: src/Components/Employee/Projects/index.js [hunk 1]
```diff
@@ -1,9 +1,49 @@
...
+
   return (
     <section className={styles.container}>
-      <h2>Projects</h2>
+      {isPending && <Loader />}
+      {error && <Error text={error} />}
+      {!isPending && !error && (
+        <Table headers={headers} data={projectsData()} title="My projects" itemsPerPage={5} />
+      )}
+      {error && <p>{error}</p>}
```
Comment (elias-92) [inline src/Components/Employee/Projects/index.js:46]: this line would not be necessary.

Comment (GuidoG95) [reply src/Components/Employee/Projects/index.js:46]: done

File: src/Components/Employee/Projects/index.js [hunk 2]
```diff
@@ -1,9 +1,48 @@
...
+import Loader from 'Components/Shared/Loader';
+import Error from 'Components/Shared/Error';
 
 const EmployeeProjects = () => {
+  const { list: projectsList, isPending, error } = useSelector((state) => state.projects);
+  const dispatch = useDispatch();
+  const headers = {
+    name: 'Project Name',
+    description: 'Description',
+    clientName: 'Client Name',
```
Comment (Baldizan) [inline src/Components/Employee/Projects/index.js:15]: Does the employee need to see this data?

File: src/Components/Employee/Projects/index.js [hunk 3]
```diff
@@ -1,9 +1,48 @@
...
+    for (let i = 0; i < projectsList.length; i++) {
+      for (let j = 0; j < projectsList[i].employees.length; j++) {
+        if (projectsList[i].employees[j].id._id === mockedEmployeeLogged) {
+          projectsList[i].startDate = projectsList[i].startDate.slice(0, 10);
+          projectsList[i].endDate = projectsList[i].endDate.slice(0, 10);
+          employeeProjects.push(projectsList[i]);
+        }
+      }
+    }
+    return employeeProjects;
```
Comment (Baldizan) [inline src/Components/Employee/Projects/index.js:36]: Shouldn't this be a map?

Comment (GuidoG95) [reply src/Components/Employee/Projects/index.js:36]: I talked abaout this in todays demo, Naila told me to leave it like this beacuse I couldn't map an array of an array to filter the data I needed

**Review by Baldizan — APPROVED** [review]

**Review by elias-92 — APPROVED** [review]

File: src/Components/Employee/Projects/index.js [hunk 4]
```diff
@@ -1,9 +1,46 @@
...
+  useEffect(() => {
+    dispatch(getProjects());
+  }, []);
+
+  const mockedEmployeeLogged = '63756ccf76cb20ad5433fa22';
+  const projectsData = () => {
+    let employeeProjects = [];
+    for (let i = 0; i < projectsList.length; i++) {
+      for (let j = 0; j < projectsList[i].employees.length; j++) {
+        if (projectsList[i].employees[j].id._id === mockedEmployeeLogged) {
```
Comment (Baldizan) [inline src/Components/Employee/Projects/index.js:27]: to fix QA's correction:
`projectsList[i].employees[j]?.id?._id`

Comment (GuidoG95) [reply src/Components/Employee/Projects/index.js]: done

**Review by mr-valen — APPROVED** [review]
Everything OK. There is a problem with **employee model** when adding multiple projects, it only accepts one project, but it's not strictly related to this PR.

**Pull request merged**